llvm/lib/CodeGen/StackProtector.cpp

```diff
@@ -176,8 +176,9 @@ static bool HasAddressTaken(const Instruction *AI, TypeSize AllocSize,
     const auto *I = cast<Instruction>(U);
     // If this instruction accesses memory make sure it doesn't access beyond
     // the bounds of the allocated object.
+    // TODO: TypeSize::getFixed should be modified to adapt to scalable vectors
```

Comment:
Isn't this just a matter of replacing `TypeSize::getFixed(MemLoc->Size.getValue())` with plain `MemLoc->Size.getValue()` (which already is a TypeSize now)?

Reply:
yes agreed